### PR TITLE
docs: Migrate link formats

### PR DESCRIPTION
**Pull request by ashleemboyer**
## Relevant links

- Preview URL: https://waypoint-git-docs-ambmigrate-link-formats-hashicorp.vercel.app/

## What

- Migrates the formats of links in Markdown content as part of the Docs Content Link Rewrites project.
  - See the one-pager for additional context on the project: [Dev Dot - Docs Content Link Rewrites](https://docs.google.com/document/d/1WaSyvoVPS-YCNiSPX0ynGpvc1gySEcpbrRoQVhimQCA/edit#heading=h.wx39rhxfetpg)
  - Most MDX files were changed by running the new [`scripts/docs-content-link-rewrites/rewrite-links`](https://github.com/hashicorp/dev-portal/blob/main/scripts/docs-content-link-rewrites/rewrite-links.ts) script in dev-portal on a local copy of the repo.
  - The `go` files changed in this PR are for updating the automatically generated documentation and associated JSON.
- Adds two new workflows, which call workflows that live in dev-portal
  - `.github/workflows/ test-link-rewrites.yml` -- this workflow is only intended for this current migration PR. The workflow it calls does a check for this, and will only run for branches named `docs/amb.migrate-link-formats`.
  - `.github/workflows/check-legacy-links-format.yml` -- this workflow is intended to be used to maintain the new link format. It does not yet check nav-data JSON files, but it will after some platform updates are made. At that time, no changes will be needed in this file since the workflow lives in dev-portal.

## Testing/Reviewing

- [ ] The following files need manual review the most:
  - [ ] `builtin/k8s/apply/platform.go`
  - [ ] `builtin/k8s/helm/platform.go`
  - [ ] `builtin/nomad/canary/releaser.go`
  - [ ] `builtin/nomad/jobspec/platform.go`
  - [ ] `builtin/vault/config_sourcer.go`
  - [ ] `internal/cli/app_docs.go`
- [ ] _**All**_ changes in this PR have automated tests associated with them, so make sure the following checks have passed:
  - [ ] All of the existing checks that were in place before this PR
  - [ ] The new [check-legacy-links-format](https://github.com/hashicorp/waypoint/actions/workflows/check-legacy-links-format.yml)
  - [ ] The new [test-link-rewrites](https://github.com/hashicorp/waypoint/actions/workflows/test-link-rewrites.yml)

### CODE REVIEW

File: website/content/docs/app-config/scope.mdx [hunk 1]
```diff
@@ -53,7 +53,7 @@ The `workspace` stanza does not support non-exact workspace matching. However,
 you can use the `label` stanza documented below to do this. All workspaces
 automatically are exposed via the `waypoint/workspace` label and label
 selectors support
-[Consul style matching operators](https://www.consul.io/api-docs/features/filtering#matching-operators)
```
Comment (ashleemboyer) [inline website/content/docs/app-config/scope.mdx:56]: Example of a case not handled yet: node URL having a hash at the end

File: website/content/docs/app-config/dynamic.mdx [hunk 2]
```diff
@@ -5,6 +5,8 @@ description: |-
   Waypoint can sync application configuration values with external systems such as Kubernetes ConfigMaps, HashiCorp Vault, Amazon Parameter Store, and more. As the values change, Waypoint will automatically update your deployments and restart your application.
 ---
 
+<!-- TODO: REMOVE COMMENT BEFORE MERGING -->
+
```
Comment (thiskevinwang) [inline website/content/docs/app-config/dynamic.mdx:9]:
```suggestion
```

**Review by topfunky — APPROVED** [review]
👍🏻

**Review by brkalow — APPROVED** [review]
👏

**Review by krantzinator — APPROVED** [review]
Thank you, this was quite the effort 🎉

**Pull request merged**